subprojects/docs/src/snippets/testing/test-suite-multi-configure-each-extracted/kotlin/build.gradle.kts

```
@@ -0,0 +1,68 @@
+import org.gradle.api.internal.tasks.testing.junit.JUnitTestFramework
+
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins {
+    `java-library`
+}
+
+version = "1.0.0"
+group = "org.gradle.sample"
+
+repositories {
+    mavenCentral()
+}
+
+// tag::multi-configure[]
+testing {
+    suites {
+        val applyMockito = { suite: JvmTestSuite -> // <1>
+            suite.useJUnitJupiter()
+            suite.dependencies {
+                implementation("org.mockito:mockito-junit-jupiter:4.6.1")
+            }
+        }
+
+        val test by getting(JvmTestSuite::class) {
+            applyMockito(this) // <2>
+        }
+
+        val integrationTest by registering(JvmTestSuite::class)
+        applyMockito(integrationTest.get()) // <3>
+
+        val functionalTest by registering(JvmTestSuite::class) {
+            useJUnit()
+            dependencies {
+                implementation("org.apache.commons:commons-lang3:3.11")
+            }
+        }
+    }
+}
+// end::multi-configure[]
+
+val checkDependencies by tasks.registering {
+    dependsOn(testing.suites.getByName("test"), testing.suites.getByName("integrationTest"), testing.suites.getByName("functionalTest"))
```

This is closer to what you want:
```suggestion
        dependsOn(
            configurations.getByName("testRuntimeClasspath"),
            configurations.getByName("integrationTestRuntimeClasspath"),
            configurations.getByName("functionalTestRuntimeClasspath")
        )
```

We want the configurations resolved before this task is called.

---

Why do we care about this?  Is it a best practice to not resolve during a task action?  I'm just asserting that when they are resolved, they contain what we expect, it doesn't matter to the test when it happens.

---

It doesn't make a difference in the correctness of the checkDependencies task AFAIK.

It's a bad practice to use configurations in a task action that you don't depend on directly.  In this case, it doesn't change anything because the task depends on the Test task and the Test task has a dependency on the same configurations.

If you had two Test tasks and only depended on one of them and used the configurations of the other one, we'd see resolution during task execution. With what we're checking for, that works. But if you had composite builds, things wouldn't work out so well.